test/integration/cycles.toml

```diff
@@ -28,88 +28,69 @@ definitions = '''
-		  "isTypedef": false,
-		  "staticSize": 8,
-		  "dynamicSize": 32,
-		  "members": [
-			{
-			  "name": "",
-			  "typePath": "",
-			  "typeName": "RawNode",
-			  "isTypedef": false,
-			  "staticSize": 16,
-			  "dynamicSize": 16,
-			  "members": [
-				{
-				  "name": "value",
-				  "typePath": "value",
-				  "typeName": "uint64_t",
-				  "staticSize": 8,
-				  "dynamicSize": 0
-				},
-				{
-				  "name": "next",
-				  "typePath": "next",
-				  "typeName": "struct RawNode *",
-				  "isTypedef": false,
-				  "staticSize": 8,
-				  "dynamicSize": 16,
-				  "members": [
-					{
-					  "name": "",
-					  "typePath": "",
-					  "typeName": "RawNode",
-					  "isTypedef": false,
-					  "staticSize": 16,
-					  "dynamicSize": 0,
-					  "members": [
-						{
-						  "name": "value",
-						  "typePath": "value",
-						  "typeName": "uint64_t",
-						  "staticSize": 8,
-						  "dynamicSize": 0
-						},
-						{
-						  "name": "next",
-						  "typePath": "next",
-						  "typeName": "struct RawNode *",
-						  "isTypedef": false,
-						  "staticSize": 8,
-						  "dynamicSize": 0
-						}
-					  ]
-					}
-				  ]
-				}
-			  ]
-			}
-		  ]
-		}
-	  ]
-	}]
+    [
+      {
+        "typeName": "RawNode",
+        "staticSize": 16,
+        "dynamicSize": 32,
+        "members": [
+          {
+            "typeName": "uint64_t",
+            "staticSize": 8,
+            "dynamicSize": 0
+          },
+          {
+            "typeName": "struct RawNode *",
+            "staticSize": 8,
+            "dynamicSize": 32,
+            "members": [
+              {
+                "typeName": "RawNode",
+                "staticSize": 16,
+                "dynamicSize": 16,
+                "members": [
+                  {
+                    "typeName": "uint64_t",
+                    "staticSize": 8,
+                    "dynamicSize": 0
+                  },
+                  {
+                    "typeName": "struct RawNode *",
+                    "staticSize": 8,
+                    "dynamicSize": 16,
+                    "members": [
+                      {
+                        "typeName": "RawNode",
+                        "staticSize": 16,
+                        "dynamicSize": 0,
+                        "members": [
+                          {
+                            "typeName": "uint64_t",
+                            "staticSize": 8,
+                            "dynamicSize": 0
+                          },
+                          {
+                            "typeName": "struct RawNode *",
+                            "staticSize": 8,
+                            "dynamicSize": 0
+                          }
+                        ]
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
     '''
 
   [cases.unique_ptr]
+    oil_skip = "OIL is not following top-level pointers"
```

Comment:
@JakeHillion This is something that we should be able to support, right?

The test case just has a single raw pointer as the argument to be probed. All other pointers are smart pointers.

Reply:
I'm not sure calling it like that makes sense. This (the test framework) makes the argument to OIL `**T`, making the template argument `*T`. It does support (and only supports) following one pointer which you've promised is accurate. Though in hindsight I should probably change that to a reference to be clearer.

Reply:
Some more discussion in #43. In short, this isn't something we can blanket enable, particularly with the change to references. We use a lot of incomplete pointers in our tests (`void*` / `class A; A*`) and automatically dereferencing them into a reference globally causes those to all fail. We'd need some extra test configuration to decide whether to dereference a pointer. For now I think it's fine to `oil_skip` or even `oil_disable` any that don't work because of it.